cli/config/configfile/file_test.go

```diff
@@ -25,3 +27,159 @@ func TestEncodeAuth(t *testing.T) {
 		t.Fatal("AuthString encoding isn't correct.")
 	}
 }
+
+func TestProxyConfig(t *testing.T) {
+	httpProxy := "http://proxy.mycorp.com:3128"
+	httpsProxy := "https://user:password@proxy.mycorp.com:3129"
+	ftpProxy := "http://ftpproxy.mycorp.com:21"
+	noProxy := "*.intra.mycorp.com"
+	defaultProxyConfig := ProxyConfig{
+		HTTPProxy:  httpProxy,
+		HTTPSProxy: httpsProxy,
+		FTPProxy:   ftpProxy,
+		NoProxy:    noProxy,
+	}
+
+	cfg := ConfigFile{
+		Proxies: map[string]ProxyConfig{
+			"default": defaultProxyConfig,
+		},
+	}
+
+	proxyConfig := cfg.ParseProxyConfig("/var/run/docker.sock", opts.NewListOpts(nil))
+
+	if *proxyConfig["HTTP_PROXY"] != httpProxy {
+		t.Fatalf("HTTP_PROXY value is incorrect.\nGot: %s\nExpected: %s\n", *proxyConfig["HTTP_PROXY"], httpProxy)
```

Comment:
This would be a lot less verbose, and would avoid any misleading error messages by using assertions:

```go
assert.Equal(t, httpProxy, proxyConfig["HTTP_PROXY"])
...
```

You can even do it as a single assertion by creating an "expected" struct to match `proxyConfig`:

```go
assert.Equal(t, expected, proxyConfig)
```